src/Core/Client/Adapter/Psr18Adapter.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace Solarium\Core\Client\Adapter;
+
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use Solarium\Core\Client\Endpoint;
+use Solarium\Core\Client\Request;
+use Solarium\Core\Client\Response;
+use Solarium\Core\Configurable;
+use Solarium\Exception\HttpException;
+
+final class Psr18Adapter extends Configurable implements AdapterInterface
```

Comment:
What is the reason for "final" (or the advantage)?

Reply:
I think we should not open this class for extension. Its does not make sense anyway as everything is private except `public function execute(Request $request, Endpoint $endpoint): Response`.

So we should not even encourage people to extend it. They should use composition if they need to customize stuff.

Or what do you think?

Reply:
OK. Let's leave it like this and wait until someone complains ;-)